tests/BenchmarkDotNet.IntegrationTests.ManualRunning.MultipleFrameworks/MultipleFrameworksTest.cs

```diff
@@ -0,0 +1,115 @@
+        private const string BenchmarkCalled = ExpectedBenchmarkNet48;
+#elif NETCOREAPP2_0
+        private const string BenchmarkCalled = ExpectedBenchmarkNetCoreApp20;
+#elif NET7_0
+        private const string BenchmarkCalled = ExpectedBenchmarkNet70;
+#endif
+
+        public MultipleFrameworksTest(ITestOutputHelper output) : base(output) { }
+
+        private static string[] GetActualLogLines(Summary summary)
+            => GetSingleStandardOutput(summary).Where(line => line.StartsWith(Prefix)).ToArray();
+
+        [Theory]
+        [InlineData(RuntimeMoniker.Net461)]
+        [InlineData(RuntimeMoniker.Net48)]
+        [InlineData(RuntimeMoniker.NetCoreApp20)]
+        [InlineData(RuntimeMoniker.Net70)]
+        public void EachFrameworkIsRebuilt(RuntimeMoniker runtime)
+        {
+            var miniJob = Job.Default.WithStrategy(RunStrategy.Monitoring).WithWarmupCount(2).WithIterationCount(3).WithInvocationCount(1).WithUnrollFactor(1).WithId("MiniJob")
+                .WithRuntime(runtime.GetRuntime());
+            var config = CreateSimpleConfig(job: miniJob);
+
+            var summary = CanExecute<SimpleBenchmark>(config);
+
+            var actualLogLines = GetActualLogLines(summary);
+            foreach (string line in actualLogLines)
+                Output.WriteLine(line);
+
+            string expectedBenchmarkCalled = runtime switch
+            {
+                RuntimeMoniker.Net461 => ExpectedBenchmarkNet461,
+                RuntimeMoniker.Net48 => ExpectedBenchmarkNet48,
+                RuntimeMoniker.NetCoreApp20 => ExpectedBenchmarkNetCoreApp20,
+                RuntimeMoniker.Net70 => ExpectedBenchmarkNet70,
+                _ => throw new ArgumentException("Unexpected runtime: " + runtime)
+            };
+
+            string[] expectedLogLines =
+            {
+                "// ### Called: GlobalSetup",
+
+                "// ### Called: IterationSetup (1)", // MainWarmup1
+                expectedBenchmarkCalled, // MainWarmup1
+                "// ### Called: IterationCleanup (1)", // MainWarmup1
+                "// ### Called: IterationSetup (2)", // MainWarmup2
+                expectedBenchmarkCalled, // MainWarmup2
+                "// ### Called: IterationCleanup (2)", // MainWarmup2
+
+                "// ### Called: IterationSetup (3)", // MainTarget1
+                expectedBenchmarkCalled, // MainTarget1
+                "// ### Called: IterationCleanup (3)", // MainTarget1
+                "// ### Called: IterationSetup (4)", // MainTarget2
+                expectedBenchmarkCalled, // MainTarget2
+                "// ### Called: IterationCleanup (4)", // MainTarget2
+                "// ### Called: IterationSetup (5)", // MainTarget3
+                expectedBenchmarkCalled, // MainTarget3
+                "// ### Called: IterationCleanup (5)", // MainTarget3
+
+                "// ### Called: GlobalCleanup"
+            };
+
+            SmartAssert.Equal(expectedLogLines, actualLogLines);
```

Comment:
the tests that verify the printed output are often flaky, we should avoid them when possible.

In this case we could pass the expected enum as string via env var and verify the compiled value in the benchmark process:

```cs
private const string TfmEnvVarName = "TfmEnvVarName";

[Theory]
[InlineData(RuntimeMoniker.Net461)]
[InlineData(RuntimeMoniker.Net48)]
[InlineData(RuntimeMoniker.NetCoreApp20)]
[InlineData(RuntimeMoniker.Net70)]
public void EachFrameworkIsRebuilt(RuntimeMoniker runtime)
{
    var config = ManualConfig.CreateEmpty().AddJob(Job.Dry.WithRuntime(runtime.GetRuntime()).WithEnvironmentVariable(TfmEnvVarName, runtime.ToString()));
    CanExecute<ValuePerTfm>(config);
}

public class ValuePerTfm
{
    private RuntimeMoniker moniker =
#if NET461
        RuntimeMoniker.Net461;
#elif NET48
        RuntimeMoniker.Net48;
#elif NETCOREAPP2_0
        RuntimeMoniker.NetCoreApp20;
#elif NET7_0
        RuntimeMoniker.Net70;
#else
        RuntimeMoniker.NotRecognized;
#endif

    [Benchmark]
    public void ThrowWhenWrong()
    {
        if (Environment.GetEnvironmentVariable(TfmEnvVarName) != moniker.ToString())
        {
            throw new InvalidOperationException($"Has not been recompiled, the value was {Environment.GetEnvironmentVariable(TfmEnvVarName)}");
        }
    }
}
```

I know it looks hacky, but it should be stable and get us what we want.